pkg/kudoctl/cmd/diagnostics/runner.go

```diff
@@ -0,0 +1,32 @@
+package diagnostics
+
+// collector - generic interface for diagnostic data collection
+// implementors are expected to return only fatal errors and handle non-fatal ones themselves
+type collector interface {
+	collect() error
+}
```

Comment:
Wouldn't it contradict the Go practice to prefer defining interfaces where they are used?

Reply:
Hmmm, good question. My main issue with it is that I can't quickly see which interface is implemented by the `resourceCollector`, `logCollector` etc.

It would help a lot if we could use this pattern at the implementing side:

```
// Ensure collector is implemented
var _ collector = &resourceCollector{}

type resourceCollector struct {
...
```

Then there's direct link to the interface and it can stay here

Reply:
Seen this trick many times, but just can't stop liking it )
Will add, definitely.